tools/flash_shelly.py

```diff
@@ -346,9 +367,13 @@ def parse_info(device_info, action, dry_run, silent_run, mode, exclude, version,
         homekit_dlurl = f'http://rojer.me/files/shelly/{version}/shelly-homekit-{model}.zip'
       break
 
-  stock_lfw = stock_release_info['data'][device_info['stock_model']]['version'].split('/v')[1].split('@')[0]
+  stock_model_info = stock_release_info['data'][stock_model]
+  stock_lfw = parseStockVersion(stock_model_info['version'])
   if not version:
-    stock_dlurl = stock_release_info['data'][device_info['stock_model']]['url']
+    if device_info['stock_model']  == 'SHRGBW2':
+      stock_dlurl = stock_model_info['url'].replace('.zip','-'+colour_mode+'.zip')
```

Comment:
avoid using + for string formatting, use f""

Reply:
done